internal/notification/handlers/notification_worker.go

```diff
@@ -42,12 +43,54 @@ type NotificationWorker struct {
 	backOff  func(current time.Duration) time.Duration
 }
 
+func (w *NotificationWorker) NextRetry(job *river.Job[*notification.Request]) time.Time {
+	return w.now().Add(job.CreatedAt.Sub(w.now()))
+}
+
+func (w *NotificationWorker) Timeout(*river.Job[*notification.Request]) time.Duration {
+	return w.config.TransactionDuration
+}
+
+// Work implements [river.Worker].
```

Comment:
nit: could throw in a check here if we wanted 
```golang
var _ river.Worker[*notification.Request] = new(NotificationWorker)
```

Reply:
that's already checked: https://github.com/zitadel/zitadel/pull/9360/files#diff-197190d10ebe20e2fa794ce488bbf8e5733552787ce3bfdb44152e8b7b89a778R146

Reply:
yes thats already checked